### ISSUE TITLE
multiplatform error declaring expect function

### ISSUE THREAD

**Issue by wwalkingg** [JetBrains/compose-multiplatform#2816]
**Describe the bug**

```kotlin
@Composable
expect fun PlatformButton(
    onClick: () -> Unit,
    interactionSource: MutableInteractionSource = MutableInteractionSource(),
    content: @Composable RowScope.() -> Unit
)
```
this compiles without error but when I Use the remember package MutableInteractionSource(), build error occurs.
```kotlin
@Composable
expect fun PlatformButton(
    onClick: () -> Unit,
    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
    content: @Composable RowScope.() -> Unit
)
```

error : 
java.lang.Exception: IR lowering failed at: PlatformButton.kt
	at androidx.compose.compiler.plugins.kotlin.lower.ComposableFunctionBodyTransformer.visitFile
**Affected platforms**
Select one of the platforms below:
- Desktop

If the bug is Android-only, report it in the [Jetpack Compose tracker](https://issuetracker.google.com/issues/new?component=612128)

**Versions**
- Kotlin version 1.8.0: 
- Compose Multiplatform version*: 1.3.0
- OS version(s)* (required for Desktop and iOS issues):  window11
- OS architecture (x86 or arm64): x86
- JDK (for desktop issues): openjdk 17

**To Reproduce**
Steps and/or the code snippet to reproduce the behavior:
Compile this code

**Comment by dima-avdeev-jb** [comment]
@wwalkingg Thanks for a bug report!

We know this issue.
One possible workaround is to make overload function:

```Kotlin
@Composable
expect fun PlatformButton(
    onClick: () -> Unit,
    interactionSource: MutableInteractionSource,
    content: @Composable RowScope.() -> Unit
)

@Composable
fun PlatformButton(
    onClick: () -> Unit,
    content: @Composable RowScope.() -> Unit,
) = PlatformButton(
    onClick = onClick,
    interactionSource = remember { MutableInteractionSource() },
    content = content,
)
```

**Comment by dima-avdeev-jb** [comment]
Initial issue here: https://github.com/JetBrains/compose-jb/issues/2806

**Comment by dima-avdeev-jb** [comment]
Reproduce here: https://github.com/dima-avdeev-jb/reproduce-issues-2816

**Comment by wwalkingg** [comment]
> Reproduce here: https://github.com/dima-avdeev-jb/reproduce-issues-2816

This can help me skip compile error. I think this issue should be the highest priority.

**Comment by eymar** [comment]
Fixed in 1.4.0 if used with kotlin 1.8.20

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-2816">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
